mautrix_telegram/util/parallel_file_transfer.py

```diff
@@ -340,7 +341,11 @@ async def encrypted(stream):
             data = encrypted(data)
             up_mime_type = "application/octet-stream"
         content_uri = await intent.upload_media(
-            data, mime_type=up_mime_type, filename=filename, size=size if not encrypt else None
+            data,
+            mime_type=up_mime_type,
+            filename=filename,
+            size=size if not encrypt else None,
+            async_upload=async_upload,
```

Comment:
This one isn't safe to do async because the encryption is streamed

```suggestion
            async_upload=async_upload and not encrypt,
```

Reply:
Oh and there's the connection limit thing too, so maybe it should just be disabled entirely here